feat: adds google pay guide (#512)
Basis-Theory/developers.basistheory.com: docs/api/connections/google-pay.mdx

## Testing

### Testing with a Google Pay token

> This endpoint is able to be used with both `TEST` and `PRODUCTION` Google Pay payment data. To use Google Pay `TEST` payment data, set the [`TEST` environment](https://developers.google.com/pay/api/web/guides/tutorial#js-load) when generating the Google Pay payment data and use a Basis Theory Application under a [`TEST` Tenant](/docs/concepts/access-controls#what-are-tenants) for making your `/connections/google-pay/tokenize` request. `PROD` Tenants will only work with `PRODUCTION` Google Pay payment data.

Reviewer: ```suggestion
This endpoint can be used with both `TEST` and `PRODUCTION` Google Pay payment data. To use Google Pay `TEST` payment data, set the [`TEST` environment](https://developers.google.com/pay/api/web/guides/tutorial#js-load) when generating the Google Pay payment data and use a Basis Theory Application under a [`TEST` Tenant](/docs/concepts/access-controls#what-are-tenants) for making your `/connections/google-pay/tokenize` request. `PROD` Tenants will only work with `PRODUCTION` Google Pay payment data.
```
